src/rp2_common/pico_bootrom/include/pico/bootrom.h

```diff
@@ -1071,6 +1071,22 @@ static inline int rom_get_last_boot_type(void) {
  */
 int rom_add_flash_runtime_partition(uint32_t start_offset, uint32_t size, uint32_t permissions);
 
+/*! \brief  Pick A/B partition for a separate partition
+ * \ingroup pico_bootrom
+ *
+ * This will perform extra checks to prevent disrupting a main image TBYB, and return errors
+ * 
+ * Also checks that the chosen partition contained a valid image
```

Comment:
I still think it's extremely vague what this function actually **does**. Any chance you could add a more detailed description, or would the purpose of this function be more obvious if I actually understood all the partitions stuff better?

Reply:
I've added a more detailed descriptions, with an example of when this function is necessary. I didn't want to delve too into the weeds (there are comments in the function for that), but hopefully it's a bit clearer